jcl/src/java.base/share/classes/com/ibm/jit/JITHelpers.java

```diff
@@ -462,7 +462,7 @@ public char byteToCharUnsigned(byte b) {
 	public native boolean acmplt(Object lhs, Object rhs);
 
 	private static long storeBits(long dest, int width, long value, int vwidth, int offset) {
-		int offsetToModify = IS_PLATFORM_LITTLE_ENDIAN ? ((offset * vwidth) % width) : ((width - 1) - ((offset * vwidth) % width));
+		int offsetToModify = (ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN) ? ((offset * vwidth) % width) : ((width - 1) - ((offset * vwidth) % width));
```

Comment:
why is this better? having a static final with no depenencies in the JCL should be more predictable. The static final here is also a primitive type so can be constant folded unlike `nativeOrder()` which returns an object so we can only get a known object constraint. This doesn't seem like an improvement...

Reply:
This is simply avoiding duplication of logic. Referencing #4741 for the background. `ByteOrder.nativeOrder()` is a user land API, and we want such calls folded away as the typical (and only) use case is to compare against `ByteOrder.LITTLE_ENDIAN` and `ByteOrder.BIG_ENDIAN`. We already more or less did this with `JITHelpers.IS_PLATFORM_LITTLE_ENDIAN`, but in a different way.

There is no reason why we should have two nearly identical ways of determining endianness. This PR consolidates everything to add support for folding away the userland API, and as such we eat our own dogfood and use the same API in the JCL.

It is no different than the rest of the fields in classes found in `foldFinalFieldsIn`. `StringCompressionFlag` is another such example.

Reply:
There are two good reasons for keeping the code 1) it means we do not add a dependency from an OpenJ9 class onto the JCL (in a seemingly unrelated package and 2) the fact the field is a primitive allows the filed to become a constant rather than just an object constraint - this can simplify eliminating tests at low optimization levels.

Reply:
3) we don't have to respect modification of the JITHelper internal final field while we do have to do the same for ByteOrder (at least notionally) - basically it simplifies handling eliminating the field load.

Reply:
I see your point. It certainly is more complex under the hood if `ByteOrder.nativeOrder()` is used instead of `IS_PLATFORM_LITTLE_ENDIAN` in `JITHelpers`. I don't think reverting back is a deal breaker. What I would like to see is removal of the `isPlatformLittleEndian()` native method though. This logic is already duplicated for `nativeOrder` [1]. We can just reuse [1] instead (`Unsafe.isBigEndian()`).

@simonameng can you make the appropriate changes?

[1] https://github.com/eclipse/openj9/blob/83f3cbdb4c413343da35f449bb97174c618df478/runtime/jcl/common/sun_misc_Unsafe.cpp#L704-L712

Reply:
@fjeremic Sure. I'll reuse `Java_sun_misc_Unsafe_isBigEndian0` in the JITHelpers instead of `ByteOrder.nativeOrder()`

Reply:
That sounds like a reasonable compromise - use Unsafe to init the field and use the field in the JITHelpers where needed sounds like a good plan.